app/views/supervisors/edit.html.erb

```diff
@@ -28,15 +28,17 @@
 
       <div class="field form-group">
         <%= form.label :display_name %>
-        <% if policy(:user).update_supervisor_name? %>
+        <% if policy(@supervisor).update_supervisor_name? %>
           <%= form.text_field :display_name, class: "form-control" %>
         <% else %>
           <input class="form-control" type="text" placeholder="<%= @supervisor.display_name %>" readonly>
         <% end %>
       </div>
 
       <div class="actions">
-        <%= form.submit "Submit", class: "btn btn-primary" %>
+        <% if policy(@supervisor).update_supervisor_email? || policy(@supervisor).update_supervisor_name? %>
```

Comment:
Behind the scenes, these two checks are actually identical. `UserPolicy#update_supervisor_name?` actually just calls `UserPolicy#update_supervisor_email?`, but it makes sense to include them both here in this check since there is no guarantee that they will be the same thing in the future.